### ISSUE TITLE
Running Godot XR application in Standalone Mode.

### ISSUE THREAD

**Issue by smohantty** [godotengine/godot-proposals#5646]
### Describe the project you are working on

I am porting Godot4.0 beta to an HMD device to run Godot XR application in Standalone Mode ( Both XR application and XR Runtime runs in the HMD device.) 

### Describe the problem or limitation you are having in your project

 Godot XR application  creates an application  window to show the content in the HMD device and also XR runtime creates a compositor window and shows it in the HMD device. As both are trying show the content in the same display , tha XR application is hiding the final stereo window made by the XR runtime.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

We need some mode like --xr or something similar to (--headless) to stop creating the application window when running a XR application in standalone environment.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

./godot --xr --path /godot_xr_app/

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Currently I can not find any work around for this usecase.

### Is there a reason why this should be core and not an add-on in the asset library?

Not sure how else we can achieve this without adding this in core.

**Comment by Calinou** [comment]
cc @BastiaanOlij

**Comment by BastiaanOlij** [comment]
Yup, I do believe we currently have a workaround in the Android implementation triggered by setting the XR mode in the export, but we need a more structural solution to prevent setting up part of the display output.

So fully endorse this proposal and the need to start discussing what this solution should look like. I would suggest we should be able to disable output to desktop with a command like switch as suggested but also through a project setting. Even on a normal desktop you want to be able to not open up a desktop window and run the app purely through the XR compositor.

Technically the difficulty is that our setup of the GL API is tied into instancing our main window and that for many platforms a lot of the input handling goes through the windowing system as well. It may require some rejigging on various platforms to not create any windows (including our splash screen).
